src/configure

```diff
@@ -46,6 +46,15 @@
 # script, i.e. any change affecting kaldi.mk or the build system as a whole.
```

Comment:
General comment.  

We either need to re-add ATLAS support into tools at expected paths or support the canonical paths on different systems.   On Ubuntu 16.04 atlas installs to /usr/[lib|include]/atlas but we only allow $ATLASROOT/[lib|include].  We should either let the inc/lib paths be set separately or  search both $ATLASROOT/[lib|include] and $ATLASROOT[lib|include]/atlas/

Reply:
The sad thing is the includes for different algebra libs have been renamed in Ubuntu, like `atlas-cblas.h`, `openblas-cblas.h` etc. In particular, atlas has 2 headers (forgot what was the second), and one of them includes the other as `<cblas.h>`, so they cannot be used as is, without either symlinking or adding our own directory to the end of the compiler's system include path, where a file named cblas.h would be just a one-liner `#include <atlas-cblas.h>`. The `/usr/include/atlas` is "internal", the atlas-*.h includes in /usr/include use them, but they are not for the end-user consumption.

14.04 and before did not rename the file cblas.h, but rather declared package conflicts, so you could not have e. g. both libatlas-dev and libopenblas-dev installed at the same time. Now they chose a different route, but introduced a bug. The fact that it's been there unreported and unfixed for may years tells us something about the way people use ATLAS...

Reply:
I think Dan's approach is the safest: cblas headers do not depend on an implementation¹, so we can carry our own, always possible to #include.
----
¹ This is not *exactly* true, as the bitness of ints for row/column indices may vary. But in practice, it's 32 for all packaged BLAS variants, except MKL, where you have a compile/link choice between 32 and 64. Since we are unlikely to go over 4 billion columns or rows, we also use 32-bit indexing with the MKL.